hnn_core/gui/gui.py

```diff
@@ -398,7 +421,102 @@ def compose(self):
                                     self.widget_tstop, self._load_info,
                                     self.layout)
 
-        return hnn_gui
+        if not return_layout:
+            return
+        else:
+            return self.app_layout
+
+    def show(self):
+        display(self.app_layout)
+
+    def take_screenshot(self, width=None, height=None, filename=None,
+                        render=True):
+        """Embed the widget to HTML and render it through iframe.
+        Parameters
+        ----------
+        width : Optional[int]
+            The width of iframe window use to show the snapshot.
+        height : Optional[int]
+            The height of iframe window use to show the snapshot.
+        filename : Optional[int]
+            The filename of snapshot.
+        render: bool
+            Will return an IFrame object if False
+
+        Returns
+        -------
+        snapshot : An iframe snapshot object that can be rendered in notebooks.
+        """
+        self._screenshots_count += 1
+        if not filename:
+            filename = f"snapshot_{self._screenshots_count}.html"
```

Comment:
I would suggest saving to a file in the `/tmp/` directory using this library: https://docs.python.org/3/library/tempfile.html

Why do you need to keep track of the filename or screenshot count?

Reply:
Because we load static html files from disk to show the "screenshot". It's actually an `<iframe>` object with local URL. And the user may take multiple screenshots at different time, so we need a mechanism to assign each screenshot a different name so that they won't interfere with each other.

Reply:
let's use `tempfile` ... I don't want to pollute the user's directory. First make a temp directory, then store all the screenshots within that directory with filenames generated by `tempfile`

Reply:
another idea, probably even cleaner:

```py
import io
from IPython.display import HTML

html_code = io.StringIO()
embed_minimal_html(html_code, views=[self.app_layout], title='')
HTML(html_code)
```

Reply:
We now do not need static files anymore! With dataurl we can load the widget content in to the `src` attribute of the `iframe` object. See 88d67460d17cd53358c9e48779fb0bc8805bf5de

Reply:
nice!